### PR TITLE
2234 show warning exceed total seats

### PR DESCRIPTION
**Pull request by eduranperez**

### CODE REVIEW

File: app/course/directives/courseTable.js [hunk 1]
```diff
@@ -524,6 +524,10 @@ let courseTable = function ($rootScope, $timeout, CourseActionCreators, $compile
               CourseActionCreators.updateSection(section);
             }
           });
+          // Check if any overflowed sections can be updated.  If yes, update them.
+          if (sectionGroup.sections && sectionGroup.sections.length > 0){
+            CourseActionCreators.updateSection();
+          }
```
Comment (flcli) [inline app/course/directives/courseTable.js:530]: **No changes necessary as getting this to work would entail an irrational amount of work to get the project to recognize modern JS features. This is just a note.**

Optional chaining to the rescue!
```js
if( sectionGroup.sections?.length > 0 ) {
  CourseActionCreators.updateSection();
}
```
No need to short-circuit if sections is null or undefined. The `?.` operator implicitly checks for you and returns `undefined` in this case, which is falsy, so it won't update the section.

**Comment by flcli** [comment]
Looks good aside from a hiccup when overflowing a course offering with a single section.

It displays the error, but the push goes through anyway because the seat offering automatically increases to meet demand.

I'm not thrilled with that UX, but I also think it could be a smaller ticket item instead of letting it hold up this whole PR.

Anyway, good work!

File: app/course/directives/courseTable.js (hunk 1 above)
Comment (eduranperez) [reply app/course/directives/courseTable.js:530]: Dang that's cool.  I did not know that was a thing

**Comment by flcli** [comment]
> Looks good aside from a hiccup when overflowing a course offering with a single section.
> 
> It displays the error, but the push goes through anyway because the seat offering automatically increases to meet demand.
> 
> I'm not thrilled with that UX, but I also think it could be a smaller ticket item instead of letting it hold up this whole PR.
> 
> Anyway, good work!

I created a ticket for this in Jira: https://ucdavis.jira.com/jira/software/projects/RTT/boards/549/backlog?selectedIssue=RTT-16

**Pull request merged**